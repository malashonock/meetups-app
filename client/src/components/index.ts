export * from './Typography/Typography';
export * from './Button/Button';
export * from './MeetupCard/MeetupCard';
export * from './UserPreview/UserPreview';
export * from './VotesCount/VotesCount';
export * from './IconButton/IconButton';
export * from './DeleteButton/DeleteButton';
export * from './EditButton/EditButton';
export * from './Header/Header';
export * from './Tooltip/Tooltip';
export * from './InputField/InputField';
export * from './TextInput/TextInput';
export * from './TextField/TextField';
export * from './TextArea/TextArea';
export * from './HelperText/HelperText';
export * from './InputLabel/InputLabel';
export * from './Tabs/Tabs';
export * from './NavTabs/NavTabs';
export * from './Tab/Tab';
export * from './TabContent/TabContent';
export * from './TabsManager/TabsManager';
export * from './TabsIndicator/TabsIndicator';
export * from './MeetupStagesTabs/MeetupStagesTabs';
export * from './MeetupStagesTabs/MeetupTabContent/MeetupTabContent';
export * from './MeetupStagesTabs/meetupTabsMapper';
export * from './NewsCard/NewsCard';
export * from './DateTimePicker/DateTimePicker';
export * from './Stepper/Stepper';
export * from './Step/Step';
export * from './StepperProgress/StepperProgress';
export * from './ImageDropbox/ImageDropbox';
export * from './ImagePreview/ImagePreview';
export * from './ImageUploader/ImageUploader';
export * from './AuthToggle/AuthToggle';
export * from './SelectField/SelectField';
export * from './RootStoreProvider/RootStoreProvider';
export * from './LanguageSelect/LanguageSelect';
export * from './ProtectedRoute/ProtectedRoute';
export * from './AvatarGroup/AvatarGroup';
export * from './LoadingSpinner/LoadingSpinner';
export * from './Portal/Portal';
export * from './Modal/Modal';
export * from './ConfirmDialog/ConfirmDialog';
export * from './ConfirmDialogProvider/ConfirmDialogProvider';
export * from './Toast/Toast';
export * from './AlertStack/AlertStack';
