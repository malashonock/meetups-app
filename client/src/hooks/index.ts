export * from './useMeetupQuery';
export * from './useNewsArticleQuery';
export * from './useStaticFileQuery';
export * from './useAuthStore';
