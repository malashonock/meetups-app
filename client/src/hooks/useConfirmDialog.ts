import { useContext } from 'react';

import { ConfirmDialogContext, ConfirmFn } from 'components';
import { Optional } from 'types';

export const useConfirmDialog = (): ConfirmFn => {
  return useContext(ConfirmDialogContext);
};
