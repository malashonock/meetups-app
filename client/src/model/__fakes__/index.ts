export * from './user.fake';
export * from './meetup.fake';
export * from './file.fake';
