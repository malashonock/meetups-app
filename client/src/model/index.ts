export * from './meetup.model';
export * from './news.model';
export * from './user.model';
export * from './login.model';
