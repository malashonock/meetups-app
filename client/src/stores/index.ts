export * from './root.store';
export * from './auth.store';
export * from './user.store';
export * from './ui.store';
export * from './meetup.store';
export * from './news.store';
