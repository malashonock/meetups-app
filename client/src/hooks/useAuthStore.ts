import { useContext } from 'react';

import { AuthStore } from 'stores';
import { RootContext } from 'components';

export const useAuthStore = (): AuthStore => {
  return useContext(RootContext).authStore;
};
