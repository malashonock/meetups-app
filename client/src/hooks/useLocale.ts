import { useContext } from 'react';

import { UiStore, Locale } from 'stores';
import { RootContext } from 'components';
import { Optional } from 'types';

export type UseLocaleResult = [
  locale: Optional<Locale>,
  setLocale: (locale: Locale) => void,
];

export const useLocale = (): UseLocaleResult => {
  const uiStore: Optional<UiStore> = useContext(RootContext)?.uiStore;

  const locale = uiStore?.locale;

  const setLocale = (locale: Locale): void => {
    if (uiStore) {
      uiStore.locale = locale;
    }
  };

  return [locale, setLocale];
};
