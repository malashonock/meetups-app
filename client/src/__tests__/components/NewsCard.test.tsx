import { render, screen } from '@testing-library/react';

import { NewsCard } from 'components';
import { mockImageWithUrl, mockNewsArticle } from 'model/__fakes__';
import { useLocale } from 'hooks';
import { Locale } from 'stores';

// Mock useLocale and useUser hooks
jest.mock('hooks', () => {
  return {
    ...jest.requireActual('hooks'),
    useLocale: jest.fn(),
  };
});
const mockUseLocale = useLocale as jest.MockedFunction<typeof useLocale>;

beforeEach(() => {
  mockUseLocale.mockReturnValue([Locale.EN, () => undefined]);
});

afterEach(() => {
  jest.resetAllMocks();
});

describe('NewsCard', () => {
  it('should match snapshot', () => {
    const { asFragment } = render(<NewsCard news={mockNewsArticle} />);

    expect(asFragment()).toMatchSnapshot();
  });

  it('should render date, title & text, image', () => {
    render(<NewsCard news={mockNewsArticle} />);

    const date = screen.getByText('1/10/23');
    expect(date).toBeInTheDocument();

    const title = screen.getByText('Test news title');
    expect(title).toBeInTheDocument();

    const text = screen.getByText('Test news text');
    expect(text).toBeInTheDocument();

    const image = screen.getByAltText('Test news title') as HTMLImageElement;
    expect(image.src).toBe(mockImageWithUrl.url);
  });
});
