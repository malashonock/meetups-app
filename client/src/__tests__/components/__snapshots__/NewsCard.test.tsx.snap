// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewsCard should match snapshot 1`] = `
<DocumentFragment>
  <article
    class="news"
    data-testid="news-card"
  >
    <figure
      class="image"
    >
      <img
        alt="Test news title"
        src="http://localhost/test.jpg"
      />
    </figure>
    <div
      class="content"
    >
      <p
        class="typography p date"
      >
        1/10/23
      </p>
      <h2
        class="typography h2 title"
      >
        Test news title
      </h2>
      <p
        class="typography p text"
      >
        Test news text
      </p>
    </div>
  </article>
</DocumentFragment>
`;
